fabtests/prov/efa/src/rdm_remote_exit_early.c

```diff
@@ -0,0 +1,225 @@
+		return ret;
+	}
+
+	/*
+	 * The handshake procedure between server and client will happen in
+	 * either ft_exchange_keys() or ft_sync()
+	 */
+	if (post_writedata) {
+		ret = ft_exchange_keys(&remote);
+		if (ret) {
+			FT_PRINTERR("ft_exchange_keys()", -ret);
+			goto out;
+		}
+	} else {
+		ret = ft_sync_inband(false);
+		if (ret) {
+			FT_PRINTERR("ft_sync_inband", -ret);
+			goto out;
+		}
+	}
+
+	if (post_tagged)
+		ft_tag = 0xabcd;
+
+	/* client post a send/writedata and then quit */
+	if (opts.dst_addr) {
+		if (post_writedata)
+			ret = ft_post_rma(FT_RMA_WRITEDATA, tx_buf,
+					  opts.transfer_size, &remote, &tx_ctx);
+		else
+			ret = ft_post_tx(ep, remote_fi_addr, opts.transfer_size,
+					 NO_CQ_DATA, &tx_ctx);
+		printf("client exits early\n");
+	} else {
+		/* server post a recv and wait for completion, it should get an
+		 * cq error as client exit early in a long protocol
+		 */
+		if (post_rx) {
+			if (post_tagged)
+				ret = fi_trecv(ep, rx_buf, rx_size, mr_desc,
+				      FI_ADDR_UNSPEC, ft_tag, 0x0, &rx_ctx);
+			else
+				ret = fi_recv(ep, rx_buf, rx_size, mr_desc,
+				      FI_ADDR_UNSPEC, &rx_ctx);
+			printf("server posts recv\n");
+		}
+
+		ft_start();
+
+		do {
+			struct fi_cq_data_entry comp = {0};
+			struct fi_cq_err_entry comp_err = {0};
+
+			ret = fi_cq_read(rxcq, &comp, 1);
+			if (ret == 1) {
+				printf("server gets CQ entry successfully\n");
+				ret = 0;
+				goto out;
+			}
+
+			ft_stop();
+			/* When server posts a recv, we expect to
+			 * get a cq entry or cq error.
+			 * If no recv is posted, it should just
+			 * poll some cq in the timeout range
+			 * and exit.
+			 */
+			if ((end.tv_sec - start.tv_sec) > timeout) {
+				if (post_rx) {
+					fprintf(stderr, "%ds timeout expired\n",
+						timeout);
+					ret = -FI_ENODATA;
+				} else {
+					printf("server polls cq and exits\n");
+					ret = 0;
+				}
+				goto out;
+			}
+
+			if (ret == -FI_EAGAIN) {
+				continue;
+			} else if (ret == -FI_EAVAIL) {
+				ret = fi_cq_readerr(rxcq, &comp_err, FI_SEND);
+				printf("server posts fi_cq_readerr, ret = %d\n", ret);
+				if (ret < 0 && ret != -FI_EAGAIN) {
+					FT_PRINTERR("fi_cq_readerr", -ret);
+					goto out;
+				} else if (ret == 1) {
+					printf("server gets CQ err entry as expected: %d, "
+					       "%s\n",
+					       comp_err.err,
+					       fi_strerror(comp_err.err));
+					ret = 0;
+					goto out;
+				}
+			} else if (ret < 0 && ret != -FI_EAGAIN) {
+				FT_PRINTERR("fi_cq_read", -ret);
+				goto out;
+			}
+		} while (ret < 1);
+	}
+
+out:
+	ft_free_res();
+
+	return ret;
+}
+
+int main(int argc, char **argv)
+{
+	int op, ret;
+
+	opts = INIT_OPTS;
+	opts.options |= FT_OPT_SIZE;
+
+	hints = fi_allocinfo();
+	if (!hints)
+		return EXIT_FAILURE;
+
+	timeout = 5;
+	int lopt_idx = 0;
+	struct option long_opts[] = {
+		{"post-rx", no_argument, NULL, LONG_OPT_POST_RX},
+		{0, 0, 0, 0}
+	};
+	while ((op = getopt_long(argc, argv, "T" ADDR_OPTS INFO_OPTS CS_OPTS API_OPTS,
```

Comment:
This is still not what I proposed .... You already have `-o` that covers `send`, `tagged` and `writedata` in 

https://github.com/ofiwg/libfabric/blob/main/fabtests/common/shared.c#L3704-L3731
, you shouldn't need to introduce `-T` again for tagged.

after the parse, you should be able to tell which op it is using based on the hints